lib/ansible/modules/system/parted.py

```diff
@@ -0,0 +1,619 @@
+  - Linux parted tool. For a full description of the fields and the
+    options check the GNU parted manual.
+notes:
+  - When fetching information about a new disks and when the version of parted
+    installed on the system is before version 3.1, the module queries directly
+    the system to obtain disk information. In this case the units CHS and CYL
+    are not supported.
+requirements: []
+options:
+  device:
+    description: The block device (disk) where to operate.
+    required: True
+  align:
+    description: Set alignment for newly created partitions.
+    choices: ['none', 'cylinder', 'minimal', 'optimal']
+    required: False
+    default: optimal
+  number:
+    description:
+     - The number of the partition to work with or the number of the partition
+       that will be created. Required when performing any action on the disk,
+       except fetching information.
+    required: False
+  unit:
+    description:
+     - Selects the current default unit that Parted will use to display
+       locations and capacities on the disk and to interpret those given by the
+       user if they are not suffixed by an unit. When fetching information about
+       a disk, it is always recommended to specify a unit.
+    choices: [
+       's', 'B', 'KB', 'KiB', 'MB', 'MiB', 'GB', 'GiB', 'TB', 'TiB', '%', 'cyl',
+       'chs', 'compact'
+    ]
+    required: False
+    default: KiB
+  label:
+    description: Creates a new disk label.
+    choices: [
+       'aix', 'amiga', 'bsd', 'dvh', 'gpt', 'loop', 'mac', 'msdos', 'pc98',
+       'sun', ''
+    ]
+    required: False
+    default: msdos
+  part_type:
+    description:
+     - Is one of 'primary', 'extended' or 'logical' and may be specified only
+       with 'msdos' or 'dvh' partition tables. A name must be specified for a
+       'gpt' partition table. Neither part-type nor name may be used with a
+       'sun' partition table.
+    choices: ['primary', 'extended', 'logical']
+    required: False
+  part_start:
+    description:
+     - Where the partition will start as offset from the beginning of the disk,
+       that is, the "distance" from the start of the disk.
+    required: False
+    default: 0%
+  part_end :
+    description:
+     - Where the partition will end as offset from the beginning of the disk,
+       that is, the "distance" from the start of the disk.
+    required: False
+    default: 100%
+  name:
+    description:
+     - Sets the name for the partition number (GPT, Mac, MIPS and PC98 only).
+    required: False
+  flags_on:
+    description: A comma-separated list of flags to set for the partition.
+    required: False
+    default: ''
+  flags_off:
+    description: A comma-separated list of flags to clear for the partition.
+    required: False
+    default: ''
+  state:
+    description:
+     - If to create or delete a partition. If not present or empty the module
+       will return only the device information.
+    choices: ['present', 'absent', '']
```

Comment:
@bcoca Having present, absent and null seems also the wrong thing to do here IMO. I think this deserves a general think-over on how to tackle this for all modules.